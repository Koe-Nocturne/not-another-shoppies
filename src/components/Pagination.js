import { React, Fragment } from 'react';

//Bootstrap
import Button from 'react-bootstrap/Button';

//Font Awesome Import
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Pagination = ({ pagination }) => {

    const [pageCount, total, handleClick] = pagination;
    const totalPages = Math.ceil(total / 10);

    //general setup for pagination bar
    const paginationSetup = [
        { icon: 'angle-left', text: '-', onClick: handleClick, isShowing: pageCount === 1},
        { icon: '', text: `${pageCount}`  },
        { icon: '', text: '/' },
        { icon: '', text: `${totalPages}` },
        { icon: 'angle-right', text: '+', onClick: handleClick, isShowing: pageCount === totalPages }
    ];


    //This will map through the pagination array and give us the pagination bar
    const createBar = () => {
        return paginationSetup.map((obj,ind) => {
            return obj.icon ?
                (<Button size='sm' 
                    variant="outline-secondary" 
                    className={ obj.isShowing ? 'hidden' : ''} 
                    onClick={() => obj.onClick(obj.text)}
                    type='button'
                    key={`pagination-button${obj.icon}`}>
                    <FontAwesomeIcon icon={obj.icon}/>
                </Button>) :
                (<span className='paginationNumbers' key={`pagination-span-${obj.text}-${ind}`}>{obj.text}</span>);
        });
    };

    return (
        <Fragment>
            {createBar()}
        </Fragment>
    );

};

export default Pagination;