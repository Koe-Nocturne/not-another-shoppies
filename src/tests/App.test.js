import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Pagination from '../components/Pagination';
// import { mount } from 'enzyme';


import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


test('paginate page renders bar', () => {
    const component = renderer.create(<Pagination pagination={[2,50,(count) => {console.log('hello '+count)}]}/>);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

//React 17 is not compatiable with enzyme. Unable to test these out. Intital test examples I did not find this out until I was done with the project

/*
test('paginate page renders bar without arrow when page starts at one', () => {
    const wrapper = mount(<Pagination pagination={[1,50,(count) => {console.log('hello '+count)}]}/>);

    let button = wrapper.first('button');
    expect(button).hasClass('hidden');

});

test('paginate page renders bar without arrow when page starts at the last page number', () => {
    const wrapper = mount(<Pagination pagination={[1,50,(count) => {console.log('hello '+count)}]}/>);

    let button = wrapper.last('button');
    expect(button).hasClass('hidden');

});
*/