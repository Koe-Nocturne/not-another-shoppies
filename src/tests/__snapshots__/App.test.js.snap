// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`paginate page renders bar 1`] = `
Array [
  <button
    className="btn btn-outline-secondary btn-sm"
    disabled={false}
    onClick={[Function]}
    type="button"
  />,
  <span
    className="paginationNumbers"
  >
    2
  </span>,
  <span
    className="paginationNumbers"
  >
    /
  </span>,
  <span
    className="paginationNumbers"
  >
    5
  </span>,
  <button
    className="btn btn-outline-secondary btn-sm"
    disabled={false}
    onClick={[Function]}
    type="button"
  />,
]
`;
